app/controllers/cities_controller.rb

```diff
@@ -1,76 +1,45 @@
 class CitiesController < ApplicationController
+  before_filter :authenticate_user!
+  load_and_authorize_resource
   before_action :set_city, only: [:show, :edit, :update, :destroy]
+  before_action :set_all_city, only: [ :index, :create, :update, :destroy]
 
-  # GET /cities
-  # GET /cities.json
   def index
-    @cities = City.all
+     @city = City.new
   end
 
-  # GET /cities/1
-  # GET /cities/1.json
-  def show
-  end
-
-  # GET /cities/new
-  def new
-    @city = City.new
-
-    #render "/shared/cities_partial.erb.html"
-  end
-
-  # GET /cities/1/edit
   def edit
   end
 
-  # POST /cities
-  # POST /cities.json
   def create
     @city = City.new(city_params)
-
-    respond_to do |format|
-      if @city.save
-        format.html { redirect_to @city, notice: 'City was successfully created.' }
-        format.json { render :show, status: :created, location: @city }
-      else
-        format.html { render :new }
-        format.json { render json: @city.errors, status: :unprocessable_entity }
-      end
-    end
+    @city.save
   end
 
-  # PATCH/PUT /cities/1
-  # PATCH/PUT /cities/1.json
   def update
-    respond_to do |format|
-      if @city.update(city_params)
-        format.html { redirect_to @city, notice: 'City was successfully updated.' }
-        format.json { render :show, status: :ok, location: @city }
-      else
-        format.html { render :edit }
-        format.json { render json: @city.errors, status: :unprocessable_entity }
-      end
-    end
+    @city.update(city_params)
+
   end
 
-  # DELETE /cities/1
-  # DELETE /cities/1.json
   def destroy
     @city.destroy
-    respond_to do |format|
-      format.html { redirect_to cities_url, notice: 'City was successfully destroyed.' }
-      format.json { head :no_content }
-    end
+
+  end
+  def delete
+   @city = City.find(params[:city_id])
   end
```

Comment:
when i click on delete city link it goes to delete method of controller . if it exist then it renders delete city partical(delete modal) on page for delete confimation. else it show error list